docs/decisions/0022-azuresdk-streaming-api-update.md

```diff
@@ -0,0 +1,251 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: { proposed }
+date: { 2023-11-13 }
+deciders: SergeyMenshykh, markwallace-microsoft, rogerbarreto
+consulted:
+informed:
+---
+
+# How to handle streaming API Changes in Azure SDK beta 9
+
+## Context and Problem Statement
+
+Azure Sdk changes their interface for streaming API in beta 9. This change is not backward compatible and requires changes in our code.
```

Comment:
The old APIs aren't still available?

Reply:
Nope, they were removed, new classes were added representing the new streaming behavior.

Reply:
Do any other LLM SDKs expose the equivalent of the old behavior?  If the Azure SDK moved away from it, why do we think it's still right for SK consumers?